daffodil-lib/src/main/scala/org/apache/daffodil/lib/util/BitOrder.scala

```diff
@@ -109,7 +109,7 @@ object Bits {
   }
 
   def reverseBytes(bb: ByteBuffer): Unit = {
-    Assert.usage(bb.position() == 0)
+    Assert.invariant(bb.position() == 0)
```

Comment:
I'm not going to try to flag all these any more. 

If it is checking an arg, that's what Assert.usage is for. 

Perhaps we should have a synonym Assert.argCheck(...) ?